user_guides/jakartaee/src/main/jbake/content/rules-wp.adoc

```diff
@@ -440,15 +440,22 @@ the Jakarta EE 9.1 Web Profile tests and the following Technology
 Compatibility Kits (TCKs):
 
 * Jakarta Contexts and Dependency Injection 3.0
-* Jakarta Dependency Injection 2.0 
+* Jakarta Dependency Injection 2.0
 * Jakarta Bean Validation 3.0
 * Jakarta XML Binding 3.0 (If XML Binding is supported)
+* Jakarta JSON Processing 2.1
+* Jakarta JSON Binding 3.0
 
 In addition to the compatibility rules outlined in this TCK User's
 Guide, Jakarta EE 9.1 implementations must also adhere to all of the
 compatibility rules defined in the User's Guides of the aforementioned
 TCKs.
 
+EE-WP21.1 If the Jakarta EE 10 Web Profile implementation uses a runtime which has
+already been validated by the Technology Compatibility Kit,
+the Jakarta EE 10 Web Profile implementation may use result of such validation
+to claim its compliance with the Technology Compatibility Kit.
+
```

Comment:
I think this will be true for the new JSON-P + JSON-B TCKs but not certain other Standalone TCKs like the Persistence 3.1 Standalone TCK due to the Jakarta EE 10 Platform TCK having more Persistence tests that are run.

Historically, I have always seen more (Persistence Provider/Persistence Container) test failures with the Platform TCK than the Standalone Persistence TCK.

In the context of what is stated above, we do not mention Persistence but that could be an example of how this paragraph might be interpreted (IMO).

Reply:
Fair enough to consider that the text only applies to the listed TCKs.

Reply:
the new rule is 21.1, sub-part of 21 having the exact list. Do you have an idea on how to improve the wording to be more explicit?

Reply:
How about something like:

```
If the Jakarta EE 10 Web Profile implementation uses a runtime which has already been validated by the JSON-B/JSON-P Technology Compatibility Kit, the Jakarta EE 10 Web Profile implementation may use result of such validation to claim its compliance with the Technology Compatibility Kit.
```

Or:
```
If the Jakarta EE 10 Web Profile implementation uses a runtime which has already been validated by the (JSON-B/JSON-P) Technology Compatibility Kit, the Jakarta EE 10 Web Profile implementation may use result of such validation to claim its compliance with the Technology Compatibility Kit.
````

Reply:
I'd rather avoid putting in specific rules for particular specs included in the profile/platform - it could easily end up with one-to-one rule to spec mapping based on which test framework(s) the spec TCK uses (even though that is unlikely to happen given there are more specs than test frameworks :-) )

Reply:
So, if we leave the rules as stated to apply to any listed TCK, would that also apply to running the Platform TCK as well?  The Platform TCK isn't in the above list but still I would like to understand if someone forks an implementation that has already passed the Platform TCK (Web Profile), if they need to include any TCK results since all of the TCKs have already been passed against the original EE implementation?

Reply:
My understanding is that each profile defines its own rules (rules EE/rules-wp EE-WP) - hence the duplicity you can see in this PR

Reply:
Yes, but do the above EE-WP rules mean that forks of GlassFish 7.0.0 can send in an EE compatibility certification request with blank TCK results because GlassFish 7.0.0 has (in the near future) already been validated as passing?

Reply:
No, there is _and_ - platform/profile tests must be run and explicitely listed “standalone” tcks may be run.

In any case I would not consider fork of a runtime to be a runtime for which TCKs already passed unless exactly same revision to build particular runtime is used